src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetPackageFileService.cs

```diff
@@ -116,6 +115,23 @@ public NuGetPackageFileService(IServiceBroker serviceBroker, INuGetTelemetryProv
             return stream;
         }
 
+        public async ValueTask<Stream?> GetReadmeAsync(Uri readmeUri, CancellationToken cancellationToken)
+        {
+            Assumes.NotNull(readmeUri);
+
+            Stream? stream;
+            if (IsEmbeddedUri(readmeUri))
```

Comment:
Just a thought: I have an active PR https://github.com/NuGet/NuGet.Client/pull/6009 to read from a local embedded file (for icon) when it's available. Would be awesome if readmes could also be read that way. 

Is my understanding correct that with this PR, it will use the existing pattern to always use the remote URI for remote package feeds? The local GPF fallback logic may be worth considering here as well. It could be a separate PR to this one.